makelib/golang.mk

```diff
@@ -333,22 +368,27 @@ $(GOFMT):
 	@$(OK) installing gofmt$(GOFMT_VERSION)
 
 $(GOIMPORTS):
-	@$(INFO) installing goimports
-	@mkdir -p $(TOOLS_HOST_DIR)/tmp-imports || $(FAIL)
-	@GOPATH=$(TOOLS_HOST_DIR)/tmp-imports GOBIN=$(TOOLS_HOST_DIR) $(GOHOST) get -u golang.org/x/tools/cmd/goimports || rm -fr $(TOOLS_HOST_DIR)/tmp-imports || $(FAIL)
-	@rm -fr $(TOOLS_HOST_DIR)/tmp-imports
-	@$(OK) installing goimports
+	@$(MAKE) go.get GOPKG=golang.org/x/tools/cmd/goimports
 
 $(GOJUNIT):
-	@$(INFO) installing go-junit-report
-	@mkdir -p $(TOOLS_HOST_DIR)/tmp-junit || $(FAIL)
-	@GOPATH=$(TOOLS_HOST_DIR)/tmp-junit GOBIN=$(TOOLS_HOST_DIR) $(GOHOST) get github.com/jstemmer/go-junit-report || rm -fr $(TOOLS_HOST_DIR)/tmp-junit || $(FAIL)
-	@rm -fr $(TOOLS_HOST_DIR)/tmp-junit
-	@$(OK) installing go-junit-report
-
-$(GOCOVER_COBERTURA):
-	@$(INFO) installing gocover-cobertura
-	@mkdir -p $(TOOLS_HOST_DIR)/tmp-gocover-cobertura || $(FAIL)
-	@GOPATH=$(TOOLS_HOST_DIR)/tmp-gocover-cobertura GOBIN=$(TOOLS_HOST_DIR) $(GOHOST) get github.com/t-yuki/gocover-cobertura || rm -fr $(TOOLS_HOST_DIR)/tmp-covcover-cobertura || $(FAIL)
-	@rm -fr $(TOOLS_HOST_DIR)/tmp-gocover-cobertura
-	@$(OK) installing gocover-cobertura
+	@$(MAKE) go.get GOPKG=github.com/jstemmer/go-junit-report
+
+$(GOCOVER_COBERTURA): $(GOJUNIT)
+	@$(MAKE) go.get GOPKG=github.com/t-yuki/gocover-cobertura
+
+go.get:
+	@$(INFO) installing $(GOPKG)
+	$(eval GO_GET_GOPATH := $(GOPATH))
+ifneq ($(GO111MODULE),on)
+	$(eval GO_GET_GOPATH := $(TOOLS_HOST_DIR)/tmp)
```

Comment:
I'm not following why this is needed when `GO111MODULE=on`.  Why does the following not work for `dep` and `go mod`:
```
@GOPATH=$(TMP_PATH) GOBIN=$(TOOLS_HOST_DIR) $(GOHOST) get $(GOPKG) || rm $(TMP_PATH) || $(FAIL)
```

Also not that you need to use a unique `/tmp` path since make could run multiple tool installs in parallel and they will collide.

Reply:
If you specify a new GOPATH for `go modules` every time running `go get`, it would download all the packages mentioned in the immediate `go.mod` seen up towards the root folder (which will be the `go.mod` in `PRJ_DIR`). This will be very inefficient for downloading tools one by one

Reply:
the `tmp` folder is used to download `src` folders like below:
```
tmp
│
└───src
    │
    └───tool1
    │
    └───tool2
```

I put the tmp folder in `WORK_DIR`, and removed the `rm -rf tmp` command and the end of `go get`. Its cleanup is delayed until `make clean` is executed

Reply:
the tmp directory is in `$(TOOLS_HOST_DIR)/tmp)` which is `.cache/linux_amd64/tmp` or something like that. `make clean` does not remove it but `make distclean` will. 

I'm really not sure this `/tmp` optimization is worth the hassle. 1) we cache the finally binary in the .cache directory so it not built every time. 2) the prior logic downloaded the source code and did not cache and had little impact on build time.

Reply:
kept the existing functionality